src/tabixIndexedFile.ts

```diff
@@ -231,11 +216,9 @@ export default class TabixIndexedFile {
         blockStart += line.length + 1
 
         // yield if we have emitted beyond the yield limit
-        linesSinceLastYield += 1
-        if (linesSinceLastYield >= this.yieldLimit) {
+        if (last - perf.now() > 10000) {
```

Comment:
I think 10 seconds is too long to monopolize a thread completely. Making it time based is a good idea, but I think 1 second at most, or probably more like 1/2 a second

Reply:
in some sense, this code should not worry about the monopolization of main thread, and instead use generators and have the calling code determine how it wants to handle it...would almost prefer if we remove this type of code somehow

Reply:
changed to yield at 0.5s